modules/a4gBidAdapter.js

```diff
@@ -0,0 +1,78 @@
+import {registerBidder} from 'src/adapters/bidderFactory';
+import * as utils from 'src/utils';
+
+const A4G_BIDDER_CODE = 'a4g';
+const A4G_CURRENCY = 'USD';
+const A4G_DEFAULT_BID_URL = '//ads.ad4game.com/v1/bid';
+const A4G_TTL = 120;
+
+const LOCATION_PARAM_NAME = 'siteurl';
+const ID_PARAM_NAME = 'id';
+const IFRAME_PARAM_NAME = 'if';
+const ZONE_ID_PARAM_NAME = 'zoneId';
+const SIZE_PARAM_NAME = 'size';
+
+const ARRAY_PARAM_SEPARATOR = ';';
+const ARRAY_SIZE_SEPARATOR = ',';
+const SIZE_SEPARATOR = 'x';
+
+export const spec = {
+  code: A4G_BIDDER_CODE,
+  isBidRequestValid: function(bid) {
+    return bid.params && !!bid.params.zoneId;
+  },
+
+  buildRequests: function(validBidRequests) {
+    let deliveryUrl = '';
+    const idParams = [];
+    const sizeParams = [];
+    const zoneIds = [];
+
+    utils._each(validBidRequests, function(bid) {
+      if (!deliveryUrl && typeof bid.params.deliveryUrl === 'string') {
+        deliveryUrl = bid.params.deliveryUrl;
+      }
+      idParams.push(bid.placementCode);
+      sizeParams.push(bid.sizes.map(size => size.join(SIZE_SEPARATOR)).join(ARRAY_SIZE_SEPARATOR));
+      zoneIds.push(bid.params.zoneId);
+    });
+
+    if (!deliveryUrl) {
+      deliveryUrl = A4G_DEFAULT_BID_URL;
+    }
+
+    return {
+      method: 'GET',
+      url: deliveryUrl,
```

Comment:
One way to do it is to add `bidId` to this object, and set it's value to the bid id of `validBidRequests` (`bidId: validBidRequests[0] && validBidRequests[0].bidId` in that case). Then `bidId` will be available on the `request` parameter in `interpretResponse` and line 61 will work without modification. You can do it however you'd prefer though